### ISSUE TITLE
End-to-end handling of various `pg_type` array types

### ISSUE THREAD

**Issue by refset** [xtdb/crux#3607]
Tools like postgres.js can expect to be able to query for and consequently handle various `pg_type` array types. Arrays of basic scalars should ideally work, but arrays of objects or other complex types may not be possible.

---
Previously the issue title was "Support for Postgres.js with `fetch_types: true`"
Currently `fetch_types: false` is needed

**Comment by egg-juxt** [comment]
Related #3582

**Comment by jarohen** [comment]
@refset mind fleshing this out with why (if indeed we know)?

**Comment by egg-juxt** [comment]
PostgresJS fetches the oids of array types at the first connection, in order to install type parsers/serializers for those array types. We don't currently have array types in our `pg_catalog.pg_type` table, so this wouldn't actually do anything. We also don't support some of the columns in the query (see #3582). So in practice we could just document the use of `fetch_types: false`.

However, I think it's important to clarify why that initial query fails with a 'malformed-query' error. The query should actually work I think and return no rows.

**Comment by refset** [comment]
This is worth checking again now that https://github.com/xtdb/xtdb/issues/3582 seems resolved on edge/nightly/soon-b2

**Comment by FiV0** [comment]
So apart from warnings this doesn't seem to affect `postgres.js` too much. I am going to deprioritize this for the time being. In case this assessment is wrong let me know.

**Comment by refset** [comment]
Any real requirements for using array types should be unblocked by transit (as a comprehensive workaround) for now, so this is not a priority.

**Comment by egg-juxt** [comment]
Duplicate of #4629
